File: backend/app/services/rsshub_defaults.py
"""Default RSSHub mirror configuration helpers."""
from __future__ import annotations

from copy import deepcopy
from typing import Dict, List


_DEFAULT_RSSHUB_MIRRORS: List[Dict[str, object]] = [
    {
        "name": "主镜像 (官方)",
        "base_url": "https://rsshub.app",
        "priority": 1,
        "is_default": True,
        "description": "RSSHub官方镜像",
    },
    {
        "name": "备用镜像1",
        "base_url": "https://rsshub.rssforever.com",
        "priority": 2,
        "description": "RSSHub备用镜像1",
    },
    {
        "name": "备用镜像2",
        "base_url": "https://rsshub.ktachibana.party",
        "priority": 3,
        "description": "RSSHub备用镜像2",
    },
    {
        "name": "备用镜像3",
        "base_url": "https://rsshub.cskaoyan.com",
        "priority": 4,
        "description": "RSSHub备用镜像3",
    },
]


def get_default_rsshub_mirrors() -> List[Dict[str, object]]:
    """Return a deep copy of the structured default mirror definitions."""
    return deepcopy(_DEFAULT_RSSHUB_MIRRORS)


def get_default_rsshub_base_urls() -> List[str]:
    """Return the base URLs from the default mirror definitions."""
    return [mirror["base_url"] for mirror in _DEFAULT_RSSHUB_MIRRORS]


# Backwards-compatible exports for modules that still expect constants
DEFAULT_RSSHUB_MIRRORS = get_default_rsshub_mirrors()
DEFAULT_RSSHUB_BASE_URLS = get_default_rsshub_base_urls()

File: backend-node/src/services/rsshubDefaults.ts
/**
 * Default RSSHub mirror configuration
 */

export interface RSSHubMirror {
  name: string;
  base_url: string;
  priority: number;
  is_default?: boolean;
  description: string;
}

const DEFAULT_RSSHUB_MIRRORS: RSSHubMirror[] = [
  {
    name: '主镜像',
    base_url: 'https://rsshub.rssforever.com',
    priority: 1,
    is_default: true,
    description: '公开镜像，当前实测可返回 RSS 路由',
  },
  {
    name: '备用镜像1',
    base_url: 'https://rsshub.ktachibana.party',
    priority: 2,
    description: '公开镜像，当前实测可返回 RSS 路由',
  },
];

export function getDefaultRSSHubMirrors(): RSSHubMirror[] {
  return JSON.parse(JSON.stringify(DEFAULT_RSSHUB_MIRRORS));
}

export function getDefaultRSSHubBaseUrls(): string[] {
  return DEFAULT_RSSHUB_MIRRORS.map(mirror => mirror.base_url);
}
